src/wizards/validators.ts

```diff
@@ -0,0 +1,88 @@
+// ------------------ Cluster validators
+
+const BROKER_FIELD = 'Broker';
+const CLUSTER_FIELD = 'Cluster name';
+const USERNAME_FIELD = 'User name';
+
+export async function validateBroker(broker: string): Promise<string | undefined> {
+    return validateFieldRequired(BROKER_FIELD, broker);
+}
+
+export async function validateClusterName(cluster: string, existingClusterNames: string[]): Promise<string | undefined> {
+    const result = validateFieldRequired(CLUSTER_FIELD, cluster);
+    if (result) {
+        return result;
+    }
+    return validateFieldUniqueValue(CLUSTER_FIELD, cluster, existingClusterNames);
+}
+
+export async function validateAuthentificationUserName(userName: string): Promise<string | undefined> {
+    return validateFieldRequired(USERNAME_FIELD, userName);
+}
+
+// ------------------ Topic validators
+
+const TOPIC_FIELD = 'Topic name';
+const PARTITIONS_FIELD = 'Number of partitions';
+const REPLICATION_FACTOR_FIELD = 'Replication Factor';
+
+const maxNameLength = 249
+const legalChars = /^[a-zA-Z0-9\\.\\-]*$/;
+
+export async function validateTopicName(topic: string, existingTopicNames: string[]): Promise<string | undefined> {
+    // See topic name validation rule at https://github.com/apache/kafka/blob/8007211cc982d8458223e866c1ee7d94b69e0249/core/src/main/scala/kafka/common/Topic.scala#L33
+    const result = validateFieldRequired(TOPIC_FIELD, topic);
+    if (result) {
+        return result;
+    }
+    else if (topic == "." || topic == "..") {
+        return `${TOPIC_FIELD} cannot be '.' or '..'`;
+    }
+    else if (topic.length > maxNameLength) {
+        return `${TOPIC_FIELD} is illegal, can't be longer than ${maxNameLength} characters`;

```

Comment:
need to be consistent. Use cannot throughout the validation (and rest of the codebase)

Reply:
I'm sorry @fbricon I don't understand what you mean? Which fix you want I do?

Reply:
replace all `can't` occurrences with `cannot`

Reply:
Ok thanks for your feedback. Fixed.